src/adapter.jl

```diff
@@ -257,3 +257,10 @@ for (left, right) in typecombinations
         end
     end
 end
+
+# Since we define the equality of GAP objects (see above),
+# we must provide a safe `hash` method, otherwise `==` results may be wrong.
+# For example, `==` for `Set`s of GAP objects may erroneously return
+# `false` if the default `hash` is used.
+Base.hash(::GapObj) = 0
```

Comment:
The 2-argument method of `hash` must be extended instead:
```suggestion
Base.hash(::GapObj, h::UInt) = xor(h, 0x0e5c5aeb80a55fa3 % UInt)
```
I added above a magic constant, as is usually the case when writing `hash` methods, to possible improve a bit performance in some (unlikely) cases where another `hash` method is defined for another type which also would return `h` (but at this point this seems paranoid and `hash` can also be just made to return `h` here).